### ISSUE TITLE
[KEYBOARD]: Text Appointments - Focus must return to Edit button when modal dialog is closed

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#2427]
## Description

When I close the modal dialog to update my mobile phone number, focus is being returned to the body element instead of the Edit button that opened it. This should be updated to return focus to the original trigger. Screenshot attached below.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

1. As a keyboard user, I want to see the yellow focus halo when I close the Mobile Phone number dialog.
2. As a tester, I want to verify the Edit button has keyboard focus by typing `document.activeElement()` into my devTools console.

## WCAG or Vendor Guidance (optional)

* [Focus Order: Understanding SC 2.4.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

## Screenshots or Trace Logs

![Screen Shot 2019-10-14 at 3 02 04 PM](https://user-images.githubusercontent.com/934879/66779610-34357580-ee94-11e9-8002-3a2b53e6c614.png)

**Comment by lkoenigsberg** [comment]
This will move to the Authenticated Experience board for consideration as we merge account and profile to a single page and redesign. (cc/ @Justin-Pickett )

**Comment by erikphansen** [comment]
It looks like the behavior for the Direct Deposit modal is correct; when you close the modal the focus is on the edit button that spawned the modal. Looking into why that works but it doesn't work for any of the contact info modals might be helpful.

**Comment by bradl3yC** [comment]
@erikphansen I've noticed that even though using ```document.activeElement``` returns the body, just hitting tab DOES go to the next tab element - its not lost on the body somewhere.

**Comment by erikphansen** [comment]
I noticed the same thing. But it works differently (and better) with the DD modals. Any clue why? The biggest problem with the current not-100%-correct implementation is that the user can't immediately hit spacebar again to reopen the modal they just closed. So both visually and functionally it isn't quite right.

**Comment by 1Copenut** [comment]
@erikphansen Right on point with your comment here. Users should be able to re-open the modal as soon as it's closed by pressing `SPACE` or `TAB`. That's one of the core tenets of modals.

@bradl3yC Focus being returned to the `body` element can be deceiving. Technically modern browsers remember where focus was when it was lost, and returns the body element, but IE11 is bad about setting focus right after the last element on the page. It's pretty disruptive to screen reader and keyboard users on IE, so it's one of the items I review pretty closely.

**Comment by Samara-Strauss** [comment]
@bradl3yC what kind of validation does this need?

**Comment by 1Copenut** [comment]
@Samara-Strauss I'm looking for a manual test to ensure the `Edit` button (link) is receiving focus when the modal window is closed:

1. The Edit text has our yellow halo
2. Pressing `ENTER` or `SPACE` should re-open the modal without having to press `TAB`

**Comment by Samara-Strauss** [comment]
@1Copenut got it.

And VALIDATED on staging! Closing!

**Comment by 1Copenut** [comment]
@Samara-Strauss Thank you! I removed it from the master accessibility ticket list.
